### PR TITLE
Update CI and Documentation to Restore Functionality from Travis

### PR DESCRIPTION
**Pull request by egordon**
Since the move to Github Actions in (#598), some CI functionality has been lost.

This includes:
* Codecov support
* Checking formatting prior to merging

In addition, the simultaneous update to Ubuntu Focal has lead to outdated documentation and unused CI code.

This PR re-adds this lost functionality, cleans up the CI code by removing old statements, and updates the main README to reflect the current use on Ubuntu Focal.

Unfortunately, this has also necessitated an upgrade to `clang-format-10`, which is available on Ubuntu Focal.

The previous version of `clang-format-6.0` had a bug such that the output of `output-replacement-xmls` (which we used to check formatting) did not match the actual replacements made in-file.

The new `clang-format-10` adds a `--dry-run` flag which offers the exact functionality we want (i.e. error if formatting is required). 

***

**Before creating a pull request**

- [ N/A ] Document new methods and classes
- [x] Format code with `make format`

**Before merging a pull request**

- [x] Set version target by selecting a milestone on the right side
- [ ] Summarize this change in `CHANGELOG.md`
- [x] Add unit test(s) for this change

### CODE REVIEW

File: .ci/install_linux_catkin.sh
Comment (amalnanavati) [inline .ci/install_linux_catkin.sh:10]: Do we want to keep the if statements to support earlier versions of Ubuntu, or do we want to only support Focal? (Either way is fine, I just figured that documenting the preference and reasoning in this PR would be useful.)

File: .ci/install_linux_catkin.sh
Comment (amalnanavati) [inline .ci/install_linux_catkin.sh:18]: Same Q about whether we want to include if statements to maintain compatibility for earlier versions of Ubuntu or not

File: .ci/install_linux_cmake.sh
Comment (amalnanavati) [inline .ci/install_linux_cmake.sh:39]: Clearly there is some support for earlier versions of Ubuntu. So maybe we should just document which versions of Ubuntu are supported? (Maybe you already did and I haven't seen that file yet)

File: .ci/install_linux_cmake.sh
Comment (amalnanavati) [inline .ci/install_linux_cmake.sh:65]: Do we need to replace this with clang-format-10.0 (as you did in `install_linux_catkin`?)

File: README.md
Comment (amalnanavati) [inline README.md:36]: I just wanted to flag this TODO in case it got overlooked

File: README.md
Comment (amalnanavati) [inline README.md:144]: 👏

File: cmake/ClangFormat.cmake
Comment (amalnanavati) [inline cmake/ClangFormat.cmake:77]: Hmm, wouldn't we want to see the output of the format executable (to identify/fix formatting issues) as opposed to redirecting it to /dev/null?

**Review by amalnanavati — APPROVED** [review]
LGTM. A few clarifying comments but nothing that should block merging.

File: cmake/ClangFormat.cmake
Comment (egordon) [reply cmake/ClangFormat.cmake:77]: (1) any issues would be output to stderr, not stdout, so outputint stdout to /dev/null won't hide any important information from console.

(2) Having error information at this stage isn't useful, since the user isn't expected to fix these issues manually. This line tripping only means that the user neglected to run `make format` on their code.

File: README.md
Comment (egordon) [reply README.md:144]: Hi! :)

Felt comfortable with that at this point. Feel free to add yourself once you have a few hundred merged lines under your belt.

File: README.md
Comment (egordon) [reply README.md:36]: Thanks for the flag! It's captured as an issue #604 

But note that, since nobody is on OS X anymore, this is very, very low priority.

File: .ci/install_linux_cmake.sh
Comment (egordon) [reply .ci/install_linux_cmake.sh:65]: Nope, clang-format-10 is only needed for the FOCAL_CATKIN_DEBUG_CODECOV (we only need to test formatting in one job), so the `cmake` installation doesn't need it.

File: .ci/install_linux_catkin.sh
Comment (egordon) [reply .ci/install_linux_catkin.sh:10]: Let it be documented both here, in the fact that our workflows run on the ubuntu-20.04 Github-Actions machine, and the new README (which changed `Trusty` -> `Focal`), that we are indeed primarily supporting Focal (with deprecated support for Bionic until that goes end-of-life).

We can debate this in #aikido if needed, but I feel like we should generally try building to the current Ubuntu LTS and leave build support for the previous Ubuntu LTS until they go end-of-life.

File: .ci/install_linux_cmake.sh
Comment (egordon) [reply .ci/install_linux_cmake.sh:39]: Recall that these are Continuous-Integration workflows. They are only executed on the OS's that we tell them to execute on, and right now, we tell them to execute on bionic and focal. We could leave these statements in, but they will never be used again, since these versions are end-of-life (Github does maintain them in github-actions).

**Pull request merged**
